geps/gep-713/index.md

```diff
@@ -1,1157 +1,1082 @@
+  )
+
+  type ColorPolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the desired state of the policy.
+    Spec ColorPolicySpec `json:"spec"`
+
+    // Status defines the current state of the policy.
+    Status ColorPolicyStatus `json:"status,omitempty"`
+  }
+
+  type ColorPolicySpec struct {
+    // TargetRefs specify the targets of the policy by label.
+    // The following kinds are supported: …
+    // +kubebuilder:validation:MinItems=1
+    // +kubebuilder:validation:MaxItems=16
+    TargetRefs []TargetRef `json:"targetRefs"`
+
+    // rest of the spec ("spec proper")…
+  }
+
+  type TargetRef struct {
+    // Group is the group of the target object.
+    Group Group `json:"group"`
+
+    // Kind is the kind of the target object.
+    Kind Kind `json:"kind"`
+
+    // Selector is the label selector of target objects of the specified kind.
+    Selector *metav1.LabelSelector `json:"selector"`
+  }
+  ```
+</details>
+
+
+##### Cross namespace references
+
+Metaresources can opt for allowing instances to target objects across Kubernetes namespaces, in which case an optional `namespace` field MUST be defined with the target reference.
+
+Although not strictly forbidden, this is in general discouraged due to [discoverability](#the-discoverability-problem) issues and security implications. Implementations that opt for designing metaresources that allow for cross namespace references should consider supporting [ReferenceGrants](https://gateway-api.sigs.k8s.io/api-types/referencegrant/?h=referencegrant) to address the security concern.
+
+<details>
+  <summary>Implementation tip</summary>
+
+  This targeting method can be implemented in Golang by using a type such as Gateway API's [`NamespacedPolicyTargetReference`](https://pkg.go.dev/sigs.k8s.io/gateway-api/apis/v1alpha2#NamespacedPolicyTargetReference) type. E.g.:
+
+  ```go
+  package color
+
+  import (
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+    gatewayapiv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+  )
+
+  type ColorPolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the desired state of the policy.
+    Spec ColorPolicySpec `json:"spec"`
+
+    // Status defines the current state of the policy.
+    Status ColorPolicyStatus `json:"status,omitempty"`
+  }
+
+  type ColorPolicySpec struct {
+    // TargetRefs specify the targets of the policy by name.
+    // The following kinds are supported: …
+    // +listType=map
+    // +listMapKey=group
+    // +listMapKey=kind
+    // +listMapKey=namespace
+    // +listMapKey=name
+    // +kubebuilder:validation:MinItems=1
+    // +kubebuilder:validation:MaxItems=16
+    TargetRefs []gatewayapiv1alpha2.NamespacedPolicyTargetReference `json:"targetRefs"`
+
+    // rest of the spec ("spec proper")…
+  }
+  ```
+</details>
+
+#### Spanning behavior across relationships of a target
```

Comment:
You're using the "indirect" and "direct" terminology in this section, it's likely worth some kind of parallel example to show how the two approaches could work. I know you have something like that in the second paragraph, but it would help me follow if there were separate subsections showing how the two methods work.

One detail I'm missing here is that in a previous iteration of policy attachment "indirect" also meant hiearchical with defaults and overrides but I'm not seeing that here.

Reply:
Indeed, here "indirect" does not imply defaults and overrides, only that the kind ultimately affected by the metaresource is not the one you see in the `targetRefs` stanza, but something else related to that object.

Other than that, I think I kept the terminology previously adopted of "direct" and "inherited" (not "indirect", which may sound surprising but does not appear anywhere in GEP-713) though, where `Inherited ⊂ Direct` (except that now incorporating enhancements discussed in https://github.com/kubernetes-sigs/gateway-api/discussions/2927.)

Happy to split this subsection into two smaller ones if it helps.